### DISCUSSION TITLE
Register UI tweaks

### DISCUSSION THREAD

**Discussion by Deguello** [actualbudget/releases#139] [Feature ideas]
IMO, it would be nice to have some updates, or the options for them, in the account view:

- Alternating row colors
- Some sort of delimiter separating current and future transactions from past ones. since there's currently no way to hide past ones. To illustrate what I mean, and not necessarily suggest this a the solution, YNAB4 had a blue line for where today would fall in the register so it was easy to tell which transactions were in the past and which weren't.
- Possibly the ability to dim (cleared?) transactions in the past (basically gray them or the row background or something like that)
- Possibly the ability to move (hide?) the recurring/transfer/etc. icons after the payee name so the payee names are all left jusfified

I'm sure there are others, or disagreements with these, but I think spiffing up the register UI would be a fairly easy win.

**Comment by MikeBishop** [comment]
> Some sort of delimiter separating current and future transactions from past ones. since there's currently no way to hide past ones. To illustrate what I mean, and not necessarily suggest this a the solution, YNAB4 had a blue line for where today would fall in the register so it was easy to tell which transactions were in the past and which weren't.

I think this space is more complicated than that, even.  Consider the different ranges of transactions you might want to see at a glance:

- Current, normal transactions
- Past transactions which have been reconciled already
- Scheduled recurring transactions
- Scheduled one-off transactions
- Future-dated transactions which have already been entered and posted

**Reply by Deguello** [reply]
Transactions are either today or in the future (and therefore above the blue line) or they are in the past (and therefore below it). The rest of that stuff, imo, doesn't really have anything to do with delimiting the past from the future.

I don't want to see reconciled transactions at all in a normal view of my register. They're dead to me as far as the default view is concerned. Pure clutter.

I don't really care whether a transaction is scheduled, either recurring or one-off. I don't make any kind of decisions based on that status. Scheduling transactions, for me, is a data entry convenience, not a workflow decision point. I think flags would be a good way to indicate transactions of those types for people to whom that would matter.

**Reply by bpaulien** [reply]
I agree mostly that it doesn't really matter if it's scheduled or not, as long as it's a future transaction, I'd like 
a.) to have it differentiated in some way to indicate that it's in the future,
b.) have it influence the running balance into the future.

And I think as long as there's an ability to filter them easily on or off, I don't generally have a need to see reconciled transactions (Really reconciled, not just cleared) in the register, but now and again, I want to see certain transactions that might have been reconciled, so being able to turn them on and off would be beneficial.

ETA: As I've said elsewhere, I'd also like to be able to see (at least in my checking account) more than just a week in the future.  I'd like to see all scheduled transactions for at least the current month, affecting the running balance, so I can see what my balance will be at any point during the month.

**Reply by MikeBishop** [reply]
So to me, the difference between the last three really hinges on what you do with "now."  YNAB was pretty firm about not having future-dated transactions in the register (and therefore not affecting your current balance).  Actual allows them, which makes the fifth category.  Third and fourth are probably differentiated simply by the circle icon to indicate a schedule, but I don't know that Actual currently has a mechanism for the fourth.  Transactions are either rules or in the register.

**Reply by Deguello** [reply]
@bpaulien Do you want to distinguish "today" from "future" from "past" or just "past" from "not past"? I agree that the running balance should be affected by whatever is in the register. What I did in YNAB4 was, on the first, "Enter in Register now" all my scheduled transactions for the month. What I'm doing in Actual now is scheduling everything for the first (with the real date in the notes) and entering them all on the first. What I think would be cool is the ability to set a rolling window that would pull in all that stuff automatically. If I could set that window for a month and have scheduled transactions automatically entered, that would be glorious.

@MikeBishop YNAB4 allowed future transactions, which was one of the reasons it worked better for me. Preventing future transactions didn't stop me, it just added busywork. I don't want someone else's philosophy to cause me busywork. The icons for scheduled works as well as a flag, though I wish they came after the text and not before it.

**Reply by bpaulien** [reply]
I have to admit, I came to YNAB after YNAB4 was no longer an option, so I never experienced that.  So I'm used to "future" transactions not affecting the balance (of account OR categories) but I think it should be affecting at least the category.  And should definitely affect the running balance.  I guess the only thing I'm not sure I care about is how it affects the "balance" of the account, as shown on the left column.  I would imagine that you'd want that to either be "current" or maybe even "cleared", but  not necessarily reflecting future (or scheduled) transactions.

If a transaction is scheduled to happen this month, I think it should indicate that in the category, so you have that future transaction taken into account before you make a decision you'll regret because you didn't realize something else was already coming out of the category.

**Comment by bpaulien** [comment]
So, there is another scenario that hasn't been mentioned yet, what about recurring weekly payments? Say my son has a violin lesson each week and I pay $25 every Sunday. If we had scheduled transactions showing up in the register, we'd need to see 4 occurrences during the month, not just the first one.

I think ynab would only show one (the next one) but if there's more than one due by the end of the month, it would be nice to see each one show up in the register.

**Reply by Deguello** [reply]
If there were a window inside which the scheduled transactions were populated into the register, this would be handled. As you got within a month (if that was your setting) of the next violin payment, it would be entered. Seven days later, the next one would be entered. By the time you rolled over to a new calendar month, all of that new month's scheduled transactions would be there for you to see.
